lib/rucio/core/rule.py

```diff
@@ -254,28 +254,69 @@ def add_rule(dids, account, copies, rse_expression, grouping, weight, lifetime,
 
             if ask_approval:
                 new_rule.state = RuleState.WAITING_APPROVAL
-                # Block manual approval for multi-rse rules
-                if len(rses) > 1:
-                    raise InvalidReplicationRule('Ask approval is not allowed for rules with multiple RSEs')
-                if len(rses) == 1 and not did.is_open and did.bytes is not None and did.length is not None:
-                    # This rule can be considered for auto-approval:
-                    rse_attr = list_rse_attributes(rse_id=rses[0]['id'], session=session)
-                    auto_approve = False
-                    if 'auto_approve_bytes' in rse_attr and 'auto_approve_files' in rse_attr:
-                        if did.bytes < int(rse_attr.get('auto_approve_bytes')) and did.length < int(rse_attr.get('auto_approve_bytes')):
-                            auto_approve = True
-                    elif did.bytes < int(rse_attr.get('auto_approve_bytes', -1)):
-                        auto_approve = True
-                    elif did.length < int(rse_attr.get('auto_approve_files', -1)):
+                configured_algorithm, auto_approve = None, False
+                algorithm_used, auto_approve_algorithm_fn = None, None
+
+                try:
+                    configured_algorithm = config_get('rules', 'auto_approve_algorithm')
+                except (NoOptionError, NoSectionError, RuntimeError):
+                    auto_approve_algorithm_fn = None
+
+                try:
+                    AUTO_APPROVE_ALGORITHMS = {}
+                    register_policy_package_algorithms('auto_approve', AUTO_APPROVE_ALGORITHMS)
+                    auto_approve_algorithm_fn = AUTO_APPROVE_ALGORITHMS[configured_algorithm]
+                except KeyError:
+                    logger(logging.WARNING, "Auto approve algorithm %s not found in policy package", configured_algorithm)
+
+                # Evaluate the rule with the auto-approve algorithm
+                if auto_approve_algorithm_fn:
+                    rule_attributes = {
+                        'account': account,
+                        'lifetime': lifetime,
+                        'rse_expression': rse_expression,
+                        'copies': copies,
+                        'weight': weight,
+                        'grouping': grouping,
+                        'locked': locked,
+                        'comment': comment,
+                        'meta': meta,
+                        'ignore_availability': ignore_availability,
+                    }
+
+                    if auto_approve_algorithm_fn(did, rule_attributes, session):
```

Comment:
So here you want the algorithm to have a boolean return and if not fall back to the default?

Reply:
I wonder if we need that? I mean this is basically so you have a fallback to the default algorithm, but I would almost be stricter on this. If the policy one doesnt work, then there just is no auto-approval.

Reply:
We (cms) do not use the default. 

``` 
if configured_algorithm == None
# use default
```